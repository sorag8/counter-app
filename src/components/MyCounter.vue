<template>
  <h2 class="titulo">Contador App</h2>
  <div class="areaContador">
    <h3 class="texto">
      Contador: <span class="valor">{{ contador }}</span>
    </h3>
    <h3 class="texto">
      Cuadrado: <span class="valor">{{ contadorCuadrado }}</span>
    </h3>
    <div>
      <button class="btn mr-1" @click="contador--">-1</button>
      <button class="btn ml-1" @click="contador++">+1</button>
    </div>
  </div>
</template>

<script setup lang="ts">
import { useCounter } from '@/composables/useCounter';

interface Props {
  valor: number;
}

const props = defineProps<Props>();
const { contador, contadorCuadrado } = useCounter(props.valor);
</script>

<style scoped>
.titulo {
  @apply text-2xl p-5 bg-slate-700 text-slate-200 text-center;
}
.areaContador {
  @apply p-5 bg-slate-200 text-center;
}
.texto {
  @apply text-xl text-gray-800 mb-2;
}
.valor {
  @apply text-xl text-red-800;
}
.btn {
  @apply px-5 py-2 bg-cyan-700 text-slate-300 hover:bg-cyan-900 hover:text-white;
}
</style>
