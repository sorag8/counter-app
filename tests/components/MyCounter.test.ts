
import { mount } from '@vue/test-utils'
import MyCounter from "@/components/MyCounter.vue"


describe( '<MyCounter />', () => {
    test('should match snapshot', () => {
        const wrapper = mount(MyCounter, {
            props:{
                valor: 5,
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    test('renders the counter value correctly', () => {
        const value = 5;
        const wrapper = mount(MyCounter, {
            props:{
                valor: value,
            }
        });
        // expect(wrapper.find('h3').text()).toContain(`Contador: ${value}`);
        // expect(wrapper.find('[data-testid="square-label"]').text()).toContain(`Cuadrado: ${value * value}`);
        // console.log(wrapper.find('h3').text());
        const [counterLabel, squareLabel] = wrapper.findAll('h3');

        expect(counterLabel.text()).toContain(`Contador: ${value}`);
        expect(squareLabel.text()).toContain(`Cuadrado: ${value * value}`);
    });

    test('decrement the counter when -1 button is clicked', async () => {
        const value = 5;
        const wrapper = mount(MyCounter, {
            props: {
                valor: value,
            }
        });

        const [counterLabel, squareLabel] = wrapper.findAll('h3');

        const btnDecrement = wrapper.find('button');
        // console.log(btnDecrement.html());
        await btnDecrement.trigger('click');

        expect(counterLabel.text()).toContain(`Contador: ${value -1}`);
        expect(squareLabel.text()).toContain(`Cuadrado: ${(value - 1) * (value - 1)}`);

    });

    test('incement the counter when +1 button is clicked twice', async () => {
        const value = 5;
        const wrapper = mount(MyCounter, {
            props: {
                valor: value,
            }
        });
        const [counterLabel, squareLabel] = wrapper.findAll('h3');

        // console.log(wrapper.find('[data-testid="increment-button"]').html());
        const btnIncrement = wrapper.find('[data-testid="increment-button"]');
      
        await btnIncrement.trigger('click');
        await btnIncrement.trigger('click');
        
        expect(counterLabel.text()).toContain(`Contador: ${value + 2}`);
        expect(squareLabel.text()).toContain(`Cuadrado: ${(value + 2) * (value + 2)}`);
    });

    test('incement the counter when +1 button is clicked twice 2', async () => {
        const value = 5;
        const wrapper = mount(MyCounter, {
            props: {
                valor: value,
            }
        });
        const [counterLabel, squareLabel] = wrapper.findAll('h3');
        
        const [, btnIncrement] = wrapper.findAll('button');

        // console.log(btnIncrement.html());
      
        await btnIncrement.trigger('click');
        await btnIncrement.trigger('click');
        
        expect(counterLabel.text()).toContain(`Contador: ${value + 2}`);
        expect(squareLabel.text()).toContain(`Cuadrado: ${(value + 2) * (value + 2)}`);
    });
});