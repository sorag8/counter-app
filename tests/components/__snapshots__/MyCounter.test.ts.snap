// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MyCounter /> > should match snapshot 1`] = `
"<h2 data-v-0352012b="" class="titulo">Contador App</h2>
<div data-v-0352012b="" class="areaContador">
  <h3 data-v-0352012b="" class="texto"> Contador: <span data-v-0352012b="" class="valor">5</span></h3>
  <h3 data-v-0352012b="" class="texto" data-testid="square-label"> Cuadrado: <span data-v-0352012b="" class="valor">25</span></h3>
  <div data-v-0352012b=""><button data-v-0352012b="" class="btn mr-1">-1</button><button data-v-0352012b="" data-testid="increment-button" class="btn ml-1">+1</button></div>
</div>"
`;
